src/main/java/it/gov/pagopa/pu/organization/service/BrokerService.java

```diff
@@ -0,0 +1,56 @@
+package it.gov.pagopa.pu.organization.service;
+
+import it.gov.pagopa.pu.organization.dto.generated.BrokerApiKeys;
+import it.gov.pagopa.pu.organization.model.Broker;
+import it.gov.pagopa.pu.organization.repository.BrokerRepository;
+import it.gov.pagopa.pu.organization.util.AESUtils;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpServerErrorException;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+@Service("brokerService")
+@Slf4j
+public class BrokerService {
+
+  private final String brokerEncryptPassword;
+  private final BrokerRepository brokerRepository;
+
+  public BrokerService(
+    @Value("${app.brokerEncryptPassword}") String brokerEncryptPassword,
+    BrokerRepository brokerRepository) {
+    this.brokerEncryptPassword = brokerEncryptPassword;
+    this.brokerRepository = brokerRepository;
+  }
+
+  private final Map<byte[], String> apiKeyDecryptMap = Collections.synchronizedMap(new HashMap<>());
+
+  public BrokerApiKeys getBrokerApiKeys(Long brokerId){
+    Broker broker = brokerRepository.findById(brokerId).orElseThrow(() -> new HttpServerErrorException(HttpStatus.NOT_FOUND));
```

Comment:
don't throw HTTP exception on ServiceLayer.
Let's throw a business (custom) exception instead, and write a @RestControllerAdvice class to catch it and transform into an HTTP Response

Reply:
done